core/trino-main/src/main/java/io/trino/server/ui/WorkerResource.java

```diff
@@ -123,6 +130,77 @@ public Response getThreads(
         return Response.status(Status.GONE).build();
     }
 
+    @ResourceSecurity(WEB_UI)
+    @GET
+    public Response getWorkerList()
+    {
+        Set<InternalNode> activeNodes = nodeManager.getAllNodes().getActiveNodes();
+        Set<InternalNode> inactiveNodes = nodeManager.getAllNodes().getInactiveNodes();
+        Set<JsonNodeInfo> jsonNodes = new HashSet<>();
+        for (Node node : activeNodes) {
+            JsonNodeInfo jsonNode = new JsonNodeInfo(node.getNodeIdentifier(), node.getHostAndPort().getHostText(), node.getVersion(), node.isCoordinator(), ACTIVE.toString().toLowerCase(Locale.ENGLISH));
+            jsonNodes.add(jsonNode);
+        }
+        for (Node node : inactiveNodes) {
+            JsonNodeInfo jsonNode = new JsonNodeInfo(node.getNodeIdentifier(), node.getHostAndPort().getHostText(), node.getVersion(), node.isCoordinator(), INACTIVE.toString().toLowerCase(Locale.ENGLISH));
+            jsonNodes.add(jsonNode);
+        }
+        return Response.ok().entity(jsonNodes).build();
+    }
+
+    public static class JsonNodeInfo
+    {
+        private final String nodeId;
+        private final String nodeIp;
+        private final String nodeVersion;
+        private final boolean coordinator;
+        private final String state;
+
+        @JsonCreator
+        public JsonNodeInfo(@JsonProperty("nodeId") String nodeId,
+                @JsonProperty("nodeIp") String nodeIp,
+                @JsonProperty("nodeVersion") String nodeVersion,
+                @JsonProperty("coordinator") boolean coordinator,
+                @JsonProperty("state") String state)
+        {
+            this.nodeId = requireNonNull(nodeId, "nodeId is null");
+            this.nodeIp = requireNonNull(nodeIp, "nodeIp is null");
+            this.nodeVersion = requireNonNull(nodeVersion, "nodeVersion is null");
+            this.coordinator = requireNonNull(coordinator, "coordinator is null");
```

Comment:
There is no need to do the `requireNonNull` check for a primitive (`boolean`) type

Reply:
Done.